subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -325,9 +334,9 @@ static struct broadcast_assistant *get_bap_broadcast_assistant(struct bt_conn *c
 		}
 	}
 
-	if (!conn_cb_registered) {
+	if (!atomic_test_bit(registered_flags, REGISTERED_CONN_CB)) {
 		bt_conn_cb_register(&conn_cb);
-		conn_cb_registered = true;
+		atomic_set_bit(registered_flags, REGISTERED_CONN_CB);
```

Comment:
Please have another look at `aics_client`. 

It's not obvious, but it's quite important that we use `atomic_test_and_set_bit` instead of `atomic_test_bit` + `atomic_set_bit`, as the former is thread safe and the latter is not. 

So basically it's something like


```
if (atomic_test_and_set_bit(...)) { abort; }

if (err) {atomic_clear_bit(...))
```

That way we take control of the bit immediately and prevent other threads from passing with `atomic_test_bit` and we can release it in case of an error.

Reply:
Ok, now I get your point :)